pkg/flow/grpc-secrets.go

```diff
@@ -2,227 +2,129 @@ package flow
 
 import (
 	"context"
+	"time"
 
 	"github.com/direktiv/direktiv/pkg/flow/bytedata"
-	"github.com/direktiv/direktiv/pkg/flow/database"
-	"github.com/direktiv/direktiv/pkg/flow/database/recipient"
 	"github.com/direktiv/direktiv/pkg/flow/grpc"
-	secretsgrpc "github.com/direktiv/direktiv/pkg/secrets/grpc"
+	"github.com/direktiv/direktiv/pkg/refactor/core"
+	"github.com/google/uuid"
 	"google.golang.org/protobuf/types/known/emptypb"
 )
 
 func (flow *flow) Secrets(ctx context.Context, req *grpc.SecretsRequest) (*grpc.SecretsResponse, error) {
 	flow.sugar.Debugf("Handling gRPC request: %s", this())
 
-	cached := new(database.CacheData)
-
-	err := flow.database.NamespaceByName(ctx, cached, req.GetNamespace())
+	ns, err := flow.edb.NamespaceByName(ctx, req.GetNamespace())
 	if err != nil {
 		return nil, err
 	}
-
-	p, err := getPagination(req.Pagination)
+	_, store, _, rollback, err := flow.beginSqlTx(ctx)
 	if err != nil {
 		return nil, err
 	}
+	defer rollback()
 
-	namespace := cached.Namespace.ID.String()
-	name := req.GetKey()
-
-	request := &secretsgrpc.GetSecretsRequest{
-		Namespace: &namespace,
-		Name:      &name,
-	}
-
-	response, err := flow.secrets.client.GetSecrets(ctx, request)
-	if err != nil {
-		return nil, err
-	}
-
-	cpds := newCustomPaginationDataSecrets()
-	pagination := newCustomPagination(cpds)
-	for i := range response.Secrets {
-		cpds.Add(response.Secrets[i].GetName())
-	}
-
-	cx, err := pagination.Paginate(p)
+	list, err := store.Secrets().GetAll(ctx, ns.ID)
 	if err != nil {
 		return nil, err
 	}
 
 	var resp grpc.SecretsResponse
 
-	resp.Namespace = cached.Namespace.Name
+	resp.Namespace = ns.Name
 	resp.Secrets = new(grpc.Secrets)
-	resp.Secrets.PageInfo = new(grpc.PageInfo)
+	resp.Secrets.PageInfo = nil
```

Comment:
TODO? I don't know if setting this as nil could cause issues, but I suspect it will.

Reply:
In the frontend it doesn't, will add a todo to check later.